src/frontends/pytorch/src/op/squeeze.cpp

```diff
@@ -5,20 +5,67 @@
 #include "openvino/op/squeeze.hpp"
 
 #include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/gather.hpp"
+#include "openvino/op/logical_and.hpp"
+#include "openvino/op/logical_not.hpp"
+#include "openvino/op/non_zero.hpp"
+#include "openvino/op/range.hpp"
+#include "openvino/op/reduce_logical_or.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/shape_of.hpp"
 #include "utils.hpp"
 
 namespace ov {
 namespace frontend {
 namespace pytorch {
 namespace op {
 
+using namespace ov::op;
+
 OutputVector translate_squeeze(const NodeContext& context) {
     num_inputs_check(context, 1, 2);
-    auto x = context.get_input(0);
+    auto input = context.get_input(0);
     if (context.input_is_none(1)) {
-        return {context.mark_node(std::make_shared<ov::op::v0::Squeeze>(x))};
+        return {context.mark_node(std::make_shared<v0::Squeeze>(input))};
     }
-    return {context.mark_node(std::make_shared<ov::op::v0::Squeeze>(x, context.get_input(1)))};
+    // Cannot provide dimensions to ov v0::Squeeze directly due to mismatch in behavior between OV and PyTorch:
+    // If provided dimension cannot be squeezed, OV raises exception, PyTorch returns dimension unmodified.
+    auto dim_input = context.get_input(1);
+    auto const_0 = context.mark_node(v0::Constant::create(element::i64, Shape{}, {0}));
+    auto const_1 = context.mark_node(v0::Constant::create(element::i64, Shape{}, {1}));
+    auto input_shape = context.mark_node(std::make_shared<v3::ShapeOf>(input));
+    auto input_rank = context.mark_node(std::make_shared<v3::ShapeOf>(input_shape));
+    auto input_rank_s = context.mark_node(std::make_shared<v0::Squeeze>(input_rank));
```

Comment:
Please use `get_shape_rank` and i32 type where possible

Reply:
Thanks, done